src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/AsciiStringSearchValuesTeddyBase.cs

```diff
@@ -0,0 +1,586 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+using static System.Buffers.StringSearchValuesHelper;
+using static System.Buffers.TeddyHelper;
+
+namespace System.Buffers
+{
+    // For a description of the Teddy multi-substring matching algorithm,
+    // see https://github.com/BurntSushi/aho-corasick/blob/master/src/packed/teddy/README.md
```

Comment:
A detailed explanation in your own words would be better, IMHO.  It can also refer to various types / helpers in our implementation to help connect the dots.

Reply:
Also, if you keep the link, I'd suggest a permalink.